### ISSUE TITLE
Ergänze Links zu GND und Wikidata in /subjects

### ISSUE THREAD

**Issue by acka47** [hbz/nwbib#656]
Im heutigen Treffen mit @hmetzner besprochen (siehe  [Notizen](https://service-wiki.hbz-nrw.de/x/LwBdPw)).

### Anforderung

**Die Wikidata- und GND-Links, die sich in https://github.com/hbz/lobid-vocabs/blob/master/nwbib/nwbib.ttl befinden, sollen unter https://nwbib.de/subjects angezeigt werden und klickbar sein.**

### Bedingungen

- Es gibt zwei properties, mit denen die Verlinkungen hergestellt werden: `skos:narrowMatch`, `skos:exactMatch`. Die müssen im UI zunächst nicht unterschiedlich dargestellt werden.
- Wenn `skos:narrowMatch` benutzt wird, gibt es häufig je zwei, teilweise je drei Links zur GND oder Wikidata. Deshalb fällt eine einfache Lösung mit einem GND- und Wikidata-Logo, hinter dem sich der Link befindet weg. (Vgl. https://digital.ub.uni-duesseldorf.de/id/871626/ zu so einer Lösung.)

Mögliche Ansätze:
- Wir ergänzen ein Link-Symbol o.ä. hinter verlinkten Notationsstellen. Bei Klick auf das Symbol geht ein Modal auf, in dem die Links nach außen angezeigt werden.
- Alternativ ließe sich das Modal auch mit Klick auf den Namen der Notationsstelle selbst öffnen.

**Comment by acka47** [comment]
@fsteeg Kannst du bitte bei Gelegenheit den Aufwand abschätzen, damit wir der NWBib-Redaktion mitteilen können, wann mit einer Umsetzung zu rechnen ist?

**Comment by fsteeg** [comment]
Der Aufwand ist schwer zu schätzen. Wenn alles problemlos klappt sollte das relativ schnell machbar sein. Es sind aber viele Komponenten involviert (das Turtle wird als JSON-LD in eine interne Elasticsearch indexiert, als Objekthierarchie in der NWBib genutzt, diese Hierarchie wird wieder als SKOS rausgeschrieben), so dass das auch aufwändiger sein könnte. 

Ich könnte es im Laufe dieses Monats (zusammen mit #657) mal angehen, und wenn es gut klappt auch umsetzen.

> Wenn skos:narrowMatch benutzt wird [...]

Ich sehe aber (noch?) kein `narrowMatch` in https://github.com/hbz/lobid-vocabs/blob/master/nwbib/nwbib.ttl

**Comment by acka47** [comment]
> Ich sehe aber (noch?) kein narrowMatch in https://github.com/hbz/lobid-vocabs/blob/master/nwbib/nwbib.ttl

Ich hatte vergessen, https://github.com/hbz/lobid-vocabs/pull/205 zu mergen. Ist jetzt erledigt. (Vielleicht hätte ich da besser die commits squashen sollen?)

**Comment by acka47** [comment]
@HMetzner kannst du bitte mal schauen, ob das für dich so passt mit den GND-/Wikidata-Links? -> https://test.nwbib.de/subjects

**Comment by HMetzner** [comment]
@acka47 @fsteeg vielen Dank für die Umsetzung! Das funktioniert gut. Könnte man dort, wo es zwei Verlinkungen gibt (Bsp. [532050 Ärzte. Heilberufe](https://test.nwbib.de/subjects#N532050) noch in Klammern und vielleicht hellgrau ergänzen? Damit man direkt weiß, welcher Link auf welchen Eintrag verweist?
Eine zweite Sache: Es könnte von Bearbeiter*innen die Rückmeldung kommen, dass das Link-Symbol direkt neben dem Kopier-Symbol unübersichtlich und bei der Bearbeitung hinderlich ist. Adrian, ich meine, wir hatten im Gespräch überlegt den Kasten alternativ mit den Notationen zu verlinken, oder? Dass der Kasten erscheint, wenn man auf die Wörter "Ärzte,Heilberufe" klickt. Wäre das möglich? Ansonsten könnte ich erstmal eine Feedback-Runde in Münster und Düsseldorf machen und schauen, ob die Bearbeitung auch so funktioniert.

**Comment by acka47** [comment]
> Es könnte von Bearbeiter*innen die Rückmeldung kommen, dass das Link-Symbol direkt neben dem Kopier-Symbol unübersichtlich und bei der Bearbeitung hinderlich ist. Adrian, ich meine, wir hatten im Gespräch überlegt den Kasten alternativ mit den Notationen zu verlinken, oder? Dass der Kasten erscheint, wenn man auf die Wörter "Ärzte,Heilberufe" klickt. Wäre das möglich?

Ja, stimmt. Ich hatte das vorab leider nicht an @fsteeg  weitergeben, bin mir aber auch nicht mehr so sicher, ob das so sinvvoll ist. Das würde glaube ich auch automatisch bedeuten, dass beim Aufruf einer Notations-URL, z.B. https://nwbib.de/subjects#N532050, der Kasten auch direkt geöffnet ist. Das möchtest du ja nicht, oder?

**Comment by HMetzner** [comment]
Nein, genau. Es ist keine Option, dass der Kasten direkt beim Aufrufen des Links geöffnet ist.

**Comment by acka47** [comment]
> Es ist keine Option, dass der Kasten direkt beim Aufrufen des Links geöffnet ist.

Dann müssen wir das glaube ich so lassen, es sei denn, @fsteeg kennt eine Möglichkeit, das wie gewünscht umzusetzen.

**Comment by acka47** [comment]
> Das funktioniert gut. Könnte man dort, wo es zwei Verlinkungen gibt (Bsp. [532050 Ärzte. Heilberufe](https://test.nwbib.de/subjects#N532050) noch in Klammern und vielleicht hellgrau ergänzen? Damit man direkt weiß, welcher Link auf welchen Eintrag verweist?

Ich nehme an, du möchtest gerne die Labels angezeigt haben. Das geht leider nicht ohne Weiteres, weil wir die ja nicht in den SKOS-Daten haben. Da müssten wir dann jeweils die GND/lobid-GND und Wikidata on-the-fly abfragen, was ein wenig Entwicklungsaufwand bedeutet. Von daher schlagen wir vor, das erst einmal so zu deployen. Ok?

**Comment by acka47** [comment]
>  es sei denn, @fsteeg kennt eine Möglichkeit, das wie gewünscht umzusetzen.

Fabian sieht da keine Möglichkeit, so dass wir es beim Link-Symbol belassen müssen.

**Comment by HMetzner** [comment]
Okay, dann ist das so. Dann kann ich mit dem NWBib-Test-Link an die Kolleg*innen herantreten? Oder wird es direkt auf der öffentlichen NWBib-Seite verfügbar? Wenn ja, ab wann? Vielen Dank!

**Comment by acka47** [comment]
> ann kann ich mit dem NWBib-Test-Link an die Kolleg*innen herantreten? Oder wird es direkt auf der öffentlichen NWBib-Seite verfügbar? Wenn ja, ab wann?

Ist schon deployt, siehe https://nwbib.de/subjects
